internal/plan/plan.go

```diff
@@ -5,10 +5,14 @@ import (
 	"fmt"
 	"log/slog"
 	"os"
+	"sync"
+	"time"
 )
 
-type PlanHandler struct {
-	Path string `required:"True"`
+type Handler struct {
+	logger *slog.Logger
+	mu     *sync.RWMutex
+	Path   string `required:"True"`
```

Comment:
1. What is `required:"True"` for? I can't see how this struct is being read to (and in case I'm missing where this happens, I would suggest to use a different struct for that - mixing configuration objects with working entities may be confusing).

2. Can we please keep the fields private? I thought we agreed on that approach when discussing the previous PR.

Reply:
1. you're right, this is not needed
2. My instinct was to make this one public - for some reason I feel that if a value is provided by the caller to initialise an object, it's only fair that the caller should be able to get that value back again (probably overloading the terms 'public' and 'private' with too much socioeconomic baggage) - but yeah, it doesn't need to be, will change.